### DISCUSSION TITLE
Intake question

### DISCUSSION THREAD

**Discussion by honzikschenk** [team401/2024-Robot-Code#36] [Ideas]
Should we integrate the intake subsystem with the scoring one?

**Comment by honzikschenk** [comment]
@jkleiber @PChild

**Comment by charlesmackenzie** [comment]
If it's an under-bumper intake, I think it makes sense to put all intake->indexing->scoring logic in the same place. We can still put them under different IO abstractions for clarity, but they should be controlled by the same subsystem.

**Comment by jkleiber** [comment]
The advantage of separating them is that you can individually command them (i.e. not having a shooter command supercede an intake command) and its easier to maintain their state separately. For example: in an under the bumper intake you may want the intake to pushing notes away if you are carrying a note, or you might want to be intaking while warming up. There are two sets of actions at play there, so separate subsystems could make more sense. 

It's totally possible to do it as one subsystem, but then you likely have two state machines in a subsystem and you need a couple helper classes on top of the IO classes to keep things organized (sort of what the arm would have been ideally last year).

On the other hand, it adds overhead in terms of communications between subsystems to have two of them. But I don't think it's so difficult to set this up since we probably don't need to share much between the subsystems.

**Reply by honzikschenk** [reply]
Okay, let's go with the separate subsystems then.
